astropy/nddata/bitmask.py

```diff
@@ -0,0 +1,411 @@
+            **single** integer is allowed and it will be interpretted as an
+            integer bitmask. For example, instead of ``'4,8'`` one could
+            simply provide string ``'12'``.
+
+        .. note::
+
+            When ``ignore_flags`` is a `str` and when it is prepended with
+            '~', then the meaning of ``ignore_flags`` parameters will be
+            reversed: now it will be interpreted as a list of bit flags to be
+            *used* (or *not ignored*) when deciding which elements of the
+            input ``bitfield`` array are "bad". Following this convention,
+            an ``ignore_flags`` string value of ``'~0'`` would be equivalent
+            to setting ``ignore_flags=None``.
+
+        .. warning::
+
+            Because prepending '~' to a string ``ignore_flags`` is equivalent
+            to setting ``flip_bits`` to `True`, ``flip_bits`` cannot be used
+            with string ``ignore_flags`` and it must be set to `None`.
+
+    flip_bits : bool, None (Default = None)
+        Specifies whether or not to invert the bits of the bitmask either
+        supplied directly through ``ignore_flags`` parameter or built from the
+        bit flags passed through ``ignore_flags`` (only when bit flags are
+        passed as Python lists of integer bit flags). Occasionally, it may be
+        useful to *consider only specific bit flags* in the ``bitfield``
+        array when creating a boolean mask as opposite to *ignoring* specific
+        bit flags as ``ignore_flags`` behaves by default. This can be achieved
+        by inverting/flipping the bits of the bitmask created from
+        ``ignore_flags`` flags which effectively changes the meaning of the
+        ``ignore_flags`` parameter from "ignore" to "use only" these flags.
+        Setting ``flip_bits`` to `None` means that no bit flipping will be
+        performed. Bit flipping for string lists of bit flags must be
+        specified by prepending '~' to string bit flag lists
+        (see documentation for ``ignore_flags`` for more details).
+
+        .. warning::
+            This parameter can be set to either `True` or `False` **ONLY** when
+            ``ignore_flags`` is either an integer bitmask or a Python
+            list of integer bit flags. When ``ignore_flags`` is either
+            `None` or a string list of flags, ``flip_bits`` **MUST** be set
+            to `None`.
+
+    good_mask_value : int, bool (Default = True)
+        This parameter is used to derive the values that will be assigned to
+        the elements in the output boolean mask array that correspond to the
+        "good" bit fields (that are 0 after zeroing bits specified by
+        ``ignore_flags``) in the input ``bitfield`` array. When
+        ``good_mask_value`` is non-zero or `True` then values in the output
+        boolean mask array corresponding to "good" bit fields in ``bitfield``
+        will be `True` (if ``dtype`` is `numpy.bool_`) or 1 (if ``dtype`` is
```

Comment:
Looks like (when I tried building documentation locally) that sphinx isn't finding a link to `numpy.bool_`. Maybe either use double backticks or change to `numpy.bool`?

Reply:
@mwcraig  ``np.bool`` does not resolve either. However, it may be better to switch to `np.bool` due to impossibility if getting this right: `dtype : data-type (Default = numpy.bool\_)`. It seems like sphinx is having problems with trailing underscores (but not everywhere - just in a few settings).